app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -14,8 +14,15 @@ module CollectionBehavior
     included do
       validates_with HasOneTitleValidator
       self.indexer = Hyrax::CollectionIndexer
+
+      property :collection_type_gid, predicate: ::RDF::Vocab::SCHEMA.additionalType, multiple: false do |index|
+        index.as :symbol, :stored_searchable
+      end
     end
 
+    attr_accessor :collection_type
```

Comment:
Should there be validation that the `collection_type` is set and persisted?

Should the `nestable?`, `discoverable?` etc. methods return a default value (based on the column defaults)? If so, I have some ideas on implementation.

I don't think that `collection_type=` should be a method on the collection. We set the collection via the `collection_type_gid`.

```ruby
def collection_type
  if collection_type_gid.present?
    CollectionType.find(collection_type_gid) # or however we retrieve collection types
  else
    nil
  end
end
```